src/linux_mcp_server/utils/decorators.py

```diff
@@ -0,0 +1,50 @@
+"""Decorators for tool functions."""
+
+import functools
+import inspect
+import os
+
+from mcp.server.fastmcp.exceptions import ToolError
+
+
+def disallow_local_execution_in_containers(func):
+    """
+    Decorator that raises a ToolError if local execution is attempted in a container.
+
+    This decorator checks if:
+    1. The 'host' parameter is None (indicating local execution)
+    2. The process is running in a container (via the 'container' environment variable)
+
+    If both conditions are true, it raises a ToolError.
+
+    Args:
+        func: The function to decorate (must have a 'host' parameter)
+
+    Returns:
+        The wrapped function
+
+    Raises:
+        ToolError: If local execution is attempted in a container
+    """
+
+    @functools.wraps(func)
+    async def wrapper(*args, **kwargs):
+        # Get the function signature to find the 'host' parameter
+        sig = inspect.signature(func)
+        bound_args = sig.bind_partial(*args, **kwargs)
+        bound_args.apply_defaults()
+
+        # Check if 'host' parameter exists and is None
+        host_value = bound_args.arguments.get("host")
+
+        # Check if running in a container and host is None (local execution)
+        if host_value is None and os.getenv("container") is not None:
```

Comment:
I'm not sure who sets this envar, but when the `container` is empty string:
```
container=""
```
then it would be interpreted as True in this condition, which I'm not sure is desired

Reply:
That's probably a safe assumption but it might be better to check that it is both set and is a non-empty value. You can either do a truthiness check: 

```python
if host_value is None and os.getenv("container"):
```

Or check that there is a truthy value returned and that it evaluates to True.
```python
container = os.getenv("container") or None  # If unset or "", will be None
if host_value is None and container:
```